### ISSUE TITLE
[Good First Issue]: [OP CONFORMANCE][TEMPLATE] Interpolate test fails in op conformance

### ISSUE THREAD

**Issue by iefode** [openvinotoolkit/openvino#23553]
### Context

OP conformance suite is validation tool checks a plugin conditions from operation implementation status perspective.
OP conformance is based on extracted from OMZ model scope operations and graphs by `ov_subgraphs_dumper` tool. Extracted graphs are saved as IR (OpenVINO Intermediate representation) and is stored in the public share.
OP conformance suite contains the following test types:
* `OpImplCheck` validates operation support by plugin. Return `true` or `false`.
* `Inference` compare device inference results vs reference over extracted IR as a model. We generate synthetic tensors to get a result. Possible results are `passed`, `failed`, `hanged` (interrupted by timeout), `crashed` and `skipped` (just in case `-shape_mode` is misaligned with graph inputs).
* `QueryModel` check possibility of graph execution per device. Have the same status as `Inference`
* `ImportExport` exports compiled model, imports it and check that models are same. Have the same status as `Inference`

Template is a simple plugin run inference using reference implementation. It means if we will run conformance over template, we should compare `TEMPLATE` vs `TEMPLATE` results. Sometimes we have bugs inside the plugin, reference implementation or test infrastructure leads to negative test result.

### What needs to be done?

**Description**:
We have several Reshape tests failed in operation conformance over TEMPLATE plugin. Do contributor should fix them:
* `conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=()`

**How to reproduce:**
1. Build the OV using -DENABLE_TESTS=ON -DENABLE_FUNCTIONAL_TESTS=ON
2. Build `ov_op_conformance_tests` target
3. Run OP conformance to download conformance IRs and run executable file with args: 
```
python3 /openvino/src/tests/test_utils/functional_test_utils/layer_tests_summary/run_conformance.py -d=TEMPLATE --gtest_filter="conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=()"
```
>**NOTE**: `run_conformance` log contains all commands to run `ov_op_conformance_tests`! You can take them to debug!
4. Check the logs inside working directory and get the results as an example:
```
[ RUN      ] conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=()

MEM_USAGE=71868KB
[ CONFORMANCE ] Influence coefficient: 8.71821e-05
[ PLUGIN      ] `SubgraphBaseTest::compile_model()` is started
[ PLUGIN      ] `SubgraphBaseTest::compile_model()` is finished successfully. Duration is 0.0654674s
[ PLUGIN      ] `SubgraphBaseTest::get_plugin_outputs()` is started
/home/efode/repo/openvino/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:94: Failure
Exception from src/inference/src/cpp/infer_request.cpp:223:
Exception from src/core/reference/include/openvino/reference/interpolate.hpp:426:
Axes are not correct!


[  FAILED  ] conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=(), where GetParam() = ("/home/efode/repo/openvino/src/tests/test_utils/functional_test_utils/layer_tests_summary/temp/models/2023.3.0-13657-d5b0f4d2d73/operation/static/Interpolate-11/f32/0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430.xml", "") (2498 ms)


```
5. To debug the test in C++ Use the following command (just an example, refer to note in third item):
```
openvino/bin/intel64/Release/ov_op_conformance_tests --device=TEMPLATE --input_folders=openvino/src/tests/test_utils/functional_test_utils/layer_tests_summary/temp/models/conformance_ir_files.lst, --report_unique_name --output_folder="conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=()" --config_path="" --shape_mode=
```

>Important NOTE: The issue can be connected with [input generation for Gather Operation](src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp). We should generate input tensor according OV specification!

### Example Pull Requests

As an example of extending input generation per specific layer: https://github.com/openvinotoolkit/openvino/pull/20897/

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/7pVRxUwdWG) - engage in discussions, ask questions and talk to OpenVINO developers
- [How to link your Pull Request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-to-an-issue-using-the-pull-request-sidebar)
- [Conformance readme](https://github.com/openvinotoolkit/openvino/tree/master/src/tests/functional/plugin/conformance/test_runner)
- [TEMPLATE plugin](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/template/README.md)
- [OV CONTRIBUTION GUIDE](https://github.com/openvinotoolkit/openvino/tree/master/?tab=readme-ov-file#how-to-contribute)

### Contact points

@iefode

**Comment by runzhangDL** [comment]
.take

**Comment by runzhangDL** [comment]
Hi, I've created a Pull Request for this issue: https://github.com/openvinotoolkit/openvino/pull/23636. It seems that I don't have permission to link that Pull Request to this issue and please let me know if there are other things need to be done from my end.

**Comment by ho-shum-chester-law** [comment]
.take

**Comment by Aksshay88** [comment]
.take

**Comment by Aksshay88** [comment]
It seems that this issue still open and it is showing not yet assigned . And I am interested to take this issue

**Comment by runzhangDL** [comment]
I've unassigned me, you can take it again, should work.

**Comment by Aksshay88** [comment]
.take

**Comment by Aksshay88** [comment]
I have located the CMakeLists.txt and and the next process to Cmake 

cmake -DENABLE_TESTS=ON -DENABLE_FUNCTIONAL_TESTS=ON ..

# Build the project using make (or other build tool)
make --jobs=$(nproc --all)


Now could you pls explain what should be exactly done or breif me like what exactly needs to be done 

Thank you !!! 
I would be grateful for anything that has been shared 
![md](https://github.com/openvinotoolkit/openvino/assets/119944779/b6bb53ed-c65d-4cab-9485-0ae294723573)
![md2](https://github.com/openvinotoolkit/openvino/assets/119944779/ad340b84-b514-4779-8884-6aa4019a21cd)

**Comment by JJMN22** [comment]
Looking to take this issue, as it is under the "Contributors Needed" tab. Is anyone still working on it?

**Comment by iefode** [comment]
Hello, dear contributors!

Thanks for attention to our product!

I think this issue should be assigned to @runzhangDL as first contributor who took the issue!
We will provide more `first good issues` soon! :)

Thanks!

**Comment by harshalubale4** [comment]
Hi, is this issue still open?

**Comment by ShivankXD** [comment]
.take

**Comment by p-wysocki** [comment]
Hello @ShivankXD, is there anything we can do to help you with the task?

**Comment by ShivankXD** [comment]
Hi @p-wysocki  , i 've been quite busy these days , can you please unassign me from this issue . Sorry for any Inconvenience .

**Comment by mlukasze** [comment]
no worries, we will be here if you would like to try again :)

**Comment by Raghavyadav17** [comment]
.take

**Comment by Raghavyadav17** [comment]
@iefode any suggestions to approach this issue ?

**Comment by p-wysocki** [comment]
bump @iefode

**Comment by ktk-07** [comment]
.take

**Comment by ktk-07** [comment]
# Clarification Needed

TLDR: I found that Interpolate-11 linear_onnx conformance failure due to axes validation. Before proposing a fix, I would like some clarifications on the intended direction.

---

Hi @iefode @mlukasze @p-wysocki,

I managed to reproduce the failure mentioned in this issue for the following test:

```
[ RUN ] conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=()

  
MEM_USAGE=52320KB
[ CONFORMANCE ] Influence coefficient: 0.00018661
[ PLUGIN ] `SubgraphBaseTest::compile_model()` is started
[ PLUGIN ] `SubgraphBaseTest::compile_model()` is finished successfully. Duration is 0.0164823s
[ PLUGIN ] `SubgraphBaseTest::get_plugin_outputs()` is started
[ REFERENCE ] `SubgraphBaseTest::calculate_refs()` is started
[ REFERENCE ] Calculate reference in runtime
[ REFERENCE ] `SubgraphBaseTest::calculate_refs()` is started

src/tests/functional/base_func_tests/src/base/ov_subgraph.cpp:97: Failure
Exception from src/inference/src/cpp/infer_request.cpp:224:
Exception from reference/interpolate.hpp:367:
Axes are not correct!

[ FAILED ] conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=(), where GetParam() = ("/home/buntr/Projects/oss/openvino/src/tests/test_utils/functional_test_utils/layer_tests_summary/temp/models/2024.3.0-15913-e4cfd8f2edd/operation/static/Interpolate-11/f32/0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430.xml", "") (92 ms)
```

The failure occurs during reference calculation:
Exception from reference/interpolate.hpp:367:  
Axes are not correct!

So the error is thrown from the reference implementation, not from the TEMPLATE plugin execution.

---

### IR Details

After looking at the .xml, .bin and .meta file of the IR, i found the IR details below: 
From the failing IR:

- Opset11 `Interpolate`
- `mode = "linear_onnx"`
- `shape_calculation_mode = "sizes"`
- `coordinate_transformation_mode = "align_corners"`
- Input shape: `[1, 65, 65, 21]` (4D)
- `axes = [1, 2]`
- `scales_and_sizes = [513, 513]`
- Output shape: `[1, 513, 513, 21]`
    

---

### Specification Reference

According to the [OpenVINO documentation for Interpolate-11](https://docs.openvino.ai/nightly/documentation/openvino-ir-format/operation-sets/operation-specs/image/interpolate-11.html):

> Only 2D, 3D, 4D, 5D tensors with axes = {0, 1}, axes = {0, 1, 2}, axes = {2, 3}, axes = {2, 3, 4} respectively are supported for `"mode" == "linear_onnx"`.

For a 4D tensor, only `axes = {2, 3}` is documented as supported.

The failing IR uses `axes = [1, 2]`, which appears to be outside the documented supported set for `linear_onnx`.

The reference implementation validates axes and throws `"Axes are not correct!"`, which seems consistent with the current documentation. But this cause a failure in the test case when we run the tests for interpolate.

---

### Clarification Needed

Given this, I would like clarification on the intended direction:

1. **Should axes = [1,2] for 4D + linear_onnx be considered unsupported?**
    - If yes, should conformance treat this configuration as _SKIPPED_ rather than _FAILED_?
    - I believe we cannot actually support axes = [1,2] because ONNX Resize semantics operate on spatial dimensions. 
    - For 4D tensors this corresponds to H and W (axes {2,3} in NCHW layout). The failing IR uses axes [1,2], which does not correspond to spatial axes and is outside the supported configuration documented for linear_onnx.
    - Alternatively, should such IRs be filtered during IR generation?

2. **Or is the current restriction outdated?**
    - If axes = [1,2] should be supported for 4D + linear_onnx, then the reference implementation likely needs to be updated to support this configuration.
3. **Or is it possible to completely remove that IR and replace that IR with a more robust test case?**

Note: [I saw a previous attempt in PR #23636](https://github.com/openvinotoolkit/openvino/pull/23636) from a while ago, where the contributor directly made changes to the reference code in interpolate.hpp causing other test case to fail. 
I want to confirm the intended approach before making changes in reference code. If I misunderstood anything, please let me know and I can adjust my investigation accordingly.

**Comment by ktk-07** [comment]
Hi @mlukasze and @p-wysocki,

Just following up on this issue. I wanted to check if there is a preferred direction for resolving the Interpolate-11 conformance failure as discussed above. Since there hasn't been activity on this issue for a while, I wanted to check whether there is someone else who would be more appropriate to contact regarding op conformance or Interpolate behavior.
If there is someone else who would be more appropriate to contact, please let me know and I can tag them as well.

I have opened a #34587 that marks this configuration as **SKIPPED** in op conformance, since it appears to be an unsupported configuration rather than a plugin failure. The PR is mainly intended to help facilitate discussion and review a concrete proposal.

From my investigation, the failure appears to come from the reference implementation rejecting `axes = [1,2]` for a 4D tensor with `mode="linear_onnx"`, which seems consistent with the current documentation (only `{2,3}` is supported for 4D).
[interploate-11 documentation](https://docs.openvino.ai/2025/documentation/openvino-ir-format/operation-sets/operation-specs/image/interpolate-11.html)
[interpolate-4 documentation](https://docs.openvino.ai/2025/documentation/openvino-ir-format/operation-sets/operation-specs/image/interpolate-4.html)

For the tldr, I just wanted to confirm with the OpenVINO team whether the intended direction is:

- treating this IR as unsupported and skipping it in conformance which i did in my pr #34587 , or
- updating the reference implementation to support this axis pattern, or  
  - there was a similar PR #23636 which did it but causes other test failes. In addition, the documentation for interpolate does not allow for it, so I am trying to confirm with the openvino maintainers if this is the direction to take ?
- filtering/removing this IR during conformance IR generation and coming up with new conformance IR generation
- or perhaps the OpenVINO team might have a better way to approach this issue?

For full details of my findings please refer to the comment above or the reply to the comment below:

> # Clarification Needed
> TLDR: I found that Interpolate-11 linear_onnx conformance failure due to axes validation. Before proposing a fix, I would like some clarifications on the intended direction.
> 
> Hi [@iefode](https://github.com/iefode) [@mlukasze](https://github.com/mlukasze) [@p-wysocki](https://github.com/p-wysocki),
> 
> I managed to reproduce the failure mentioned in this issue for the following test:
> 
> ```
> [ RUN ] conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=()
> 
>   
> MEM_USAGE=52320KB
> [ CONFORMANCE ] Influence coefficient: 0.00018661
> [ PLUGIN ] `SubgraphBaseTest::compile_model()` is started
> [ PLUGIN ] `SubgraphBaseTest::compile_model()` is finished successfully. Duration is 0.0164823s
> [ PLUGIN ] `SubgraphBaseTest::get_plugin_outputs()` is started
> [ REFERENCE ] `SubgraphBaseTest::calculate_refs()` is started
> [ REFERENCE ] Calculate reference in runtime
> [ REFERENCE ] `SubgraphBaseTest::calculate_refs()` is started
> 
> src/tests/functional/base_func_tests/src/base/ov_subgraph.cpp:97: Failure
> Exception from src/inference/src/cpp/infer_request.cpp:224:
> Exception from reference/interpolate.hpp:367:
> Axes are not correct!
> 
> [ FAILED ] conformance_Interpolate/ReadIRTest.Inference/Op=Interpolate.11_Type=f32_Shape=static_IR=0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430_Device=TEMPLATE_Config=(), where GetParam() = ("/home/buntr/Projects/oss/openvino/src/tests/test_utils/functional_test_utils/layer_tests_summary/temp/models/2024.3.0-15913-e4cfd8f2edd/operation/static/Interpolate-11/f32/0f670e49f962b0a7abc6b4f1fbf9592db592a6a78eb3e083dd4027b9f9607430.xml", "") (92 ms)
> ```
> 
> The failure occurs during reference calculation: Exception from reference/interpolate.hpp:367: Axes are not correct!
> 
> So the error is thrown from the reference implementation, not from the TEMPLATE plugin execution.
> 
> ### IR Details
> After looking at the .xml, .bin and .meta file of the IR, i found the IR details below: From the failing IR:
> 
> * Opset11 `Interpolate`
> * `mode = "linear_onnx"`
> * `shape_calculation_mode = "sizes"`
> * `coordinate_transformation_mode = "align_corners"`
> * Input shape: `[1, 65, 65, 21]` (4D)
> * `axes = [1, 2]`
> * `scales_and_sizes = [513, 513]`
> * Output shape: `[1, 513, 513, 21]`
> 
> ### Specification Reference
> According to the [OpenVINO documentation for Interpolate-11](https://docs.openvino.ai/nightly/documentation/openvino-ir-format/operation-sets/operation-specs/image/interpolate-11.html):
> 
> > Only 2D, 3D, 4D, 5D tensors with axes = {0, 1}, axes = {0, 1, 2}, axes = {2, 3}, axes = {2, 3, 4} respectively are supported for `"mode" == "linear_onnx"`.
> 
> For a 4D tensor, only `axes = {2, 3}` is documented as supported.
> 
> The failing IR uses `axes = [1, 2]`, which appears to be outside the documented supported set for `linear_onnx`.
> 
> The reference implementation validates axes and throws `"Axes are not correct!"`, which seems consistent with the current documentation. But this cause a failure in the test case when we run the tests for interpolate.
> 
> ### Clarification Needed
> Given this, I would like clarification on the intended direction:
> 
> 1. **Should axes = [1,2] for 4D + linear_onnx be considered unsupported?**
>    
>    * If yes, should conformance treat this configuration as _SKIPPED_ rather than _FAILED_?
>    * I believe we cannot actually support axes = [1,2] because ONNX Resize semantics operate on spatial dimensions.
>    * For 4D tensors this corresponds to H and W (axes {2,3} in NCHW layout). The failing IR uses axes [1,2], which does not correspond to spatial axes and is outside the supported configuration documented for linear_onnx.
>    * Alternatively, should such IRs be filtered during IR generation?
> 2. **Or is the current restriction outdated?**
>    
>    * If axes = [1,2] should be supported for 4D + linear_onnx, then the reference implementation likely needs to be updated to support this configuration.
> 3. **Or is it possible to completely remove that IR and replace that IR with a more robust test case?**
> 
> Note: [I saw a previous attempt in PR #23636](https://github.com/openvinotoolkit/openvino/pull/23636) from a while ago, where the contributor directly made changes to the reference code in interpolate.hpp causing other test case to fail. I want to confirm the intended approach before making changes in reference code. If I misunderstood anything, please let me know and I can adjust my investigation accordingly.

Thanks!

**Comment by ktk-07** [comment]
Hi @mlukasze @p-wysocki . Just a reminder that I already submitted a PR. In case you missed it.

**Comment by ktk-07** [comment]
.release

**Comment by divyanshutata09** [comment]
.take
